pkg/lib/indexer/indexer.go

```diff
@@ -513,38 +523,72 @@ func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
 	}
 
 	var errs []error
+	var g errgroup.Group
+	var mu = &sync.Mutex{}
+
+	concurrentGoroutines := make(chan struct{}, concurrencyLimitForExport)
+
 	for _, bundleImage := range bundles {
-		// try to name the folder
-		folderName, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
-		if err != nil {
+		bundleImage := bundleImage // to ensure every go routine gets a it's own copy of closure variable
+
+		g.Go(func() error {
+			concurrentGoroutines <- struct{}{}
+			defer func() { <-concurrentGoroutines }()
+
+			// get the bundle version to be used in naming the folder
+			bundleVersion, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
+			if err != nil {
+				i.Logger.Errorf("error getting bundle version from image: %s", err)
+				return err
+			}
+			if bundleVersion == "" {
+				// operator-registry does not care about the folder name
+				bundleVersion = bundleImage
+			}
+			// fetch bundle name from bundle image
+			pkgName := strings.FieldsFunc(bundleImage, splitBundlePath)[2]
+			exporter := bundle.NewExporterForBundle(bundleImage, filepath.Join(request.DownloadPath, pkgName, bundleVersion), request.ContainerTool)
+			if err := exporter.Export(); err != nil {
+				err = fmt.Errorf("error exporting bundle from image: %s", err)
+				mu.Lock()
+				errs = append(errs, err)
+				mu.Unlock()
+			}
 			return err
-		}
-		if folderName == "" {
-			// operator-registry does not care about the folder name
-			folderName = bundleImage
-		}
-		exporter := bundle.NewExporterForBundle(bundleImage, filepath.Join(request.DownloadPath, folderName), request.ContainerTool)
-		if err := exporter.Export(); err != nil {
-			err = fmt.Errorf("error exporting bundle from image: %s", err)
-			errs = append(errs, err)
-		}
+		})
 	}
-	if err != nil {
-		errs = append(errs, err)
+
+	// Wait for all go routines to finish export
+	if err := g.Wait(); err != nil {
+		i.Logger.Errorf("Error in one of the go routines while unpacking the bundle: %s", err)
```

Comment:
Nit: I'm not convinced that logging here is helpful for users.

Reply:
+1, it should probably not refer to the implementation

Reply:
modified code doesn't use this anymore